src/app/clusters/ota-requestor/OTARequestorDriver.h

```diff
@@ -128,7 +128,9 @@ class OTARequestorDriver
     // Driver picks the OTA Provider that should be used for the next query and update. The Provider is picked according to
     // the driver's internal logic such as, for example, traversing the default providers list.
     // Returns true if there is a Provider available for the next query, returns false otherwise.
-    virtual bool GetNextProviderLocation(ProviderLocationType & providerLocation) = 0;
+    // [in] has_providers_list_been_exhausted - set to TRUE if the list of providers has been traversed until the end and has looped
+    // back to the beginning.
+    virtual bool GetNextProviderLocation(ProviderLocationType & providerLocation, bool & has_providers_list_been_exhausted) = 0;
```

Comment:
nit: cascalCase is being used through this file for function variable names
```suggestion
    virtual bool GetNextProviderLocation(ProviderLocationType & providerLocation, bool & listExhausted) = 0;
```